File: flutter/lib/domain/user/usecases/loginUseCase.dart
import '../userRepository.dart';

import '../login.dart';
import '../user.dart';

class LoginUseCase {
  UserRepository userRepository;

  LoginUseCase(UserRepository userRepository) {
    this.userRepository = userRepository;
  }

  Future<User> run(Login login) async {
    User user = await userRepository.login(login);
    await userRepository.store(user);
    return user;
  }
}

File: web/src/domain/user/usecases/LoginUseCase.ts
import UserRepository from "../UserRepository";
import User from "../User";
import Login from "../Login";

export default class LoginUseCase {
    userRepository: UserRepository;

    constructor(userRepository: UserRepository) {
        this.userRepository = userRepository;
    }

    async run(login: Login): Promise<User> {
        return await this.userRepository.login(login);
    }
}